### DISCUSSION TITLE
Show This Person Less

### DISCUSSION THREAD

**Discussion by eecue** [RhetTbull/osxphotos#935] [Q&A]
Can this data be exposed, maybe in `photo.person_info[0].show_less = True`

**Comment by RhetTbull** [comment]
Do you mean to set the flag or read it?  Both are do-able but reading it is likely easier (no database manipulation). I'd first need to know where in the database this is stored and this isn't a feature I use or have explored. My guess is it's one of the flags in `ZPERSON` or in the data blob on `ZPERSONREFERENCE`.  You can try the following if you're interested in tracking this down:

- `osxphotos snap`
- Apply "Show this person less" in Photos but make no other changes
- `osxphotos diff`

**Comment by eecue** [comment]
Just read it in python no need to write:

> Using last opened Photos library: /Users/eecue/Pictures/Photos Library Sorted.photoslibrary
> INSERT INTO ACHANGE(Z_PK,Z_ENT,Z_OPT,ZCHANGETYPE,ZENTITY,ZENTITYPK,ZTRANSACTIONID,ZCOLUMNS,ZTOMBSTONE0,ZTOMBSTONE1,ZTOMBSTONE2,ZTOMBSTONE3,ZTOMBSTONE4,ZTOMBSTONE5) VALUES(10892976,16001,NULL,1,60,9,110434,x'0300',NULL,NULL,NULL,NULL,NULL,
> INSERT INTO ACHANGE(Z_PK,Z_ENT,Z_OPT,ZCHANGETYPE,ZENTITY,ZENTITYPK,ZTRANSACTIONID,ZCOLUMNS,ZTOMBSTONE0,ZTOMBSTONE1,ZTOMBSTONE2,ZTOMBSTONE3,ZTOMBSTONE4,ZTOMBSTONE5) VALUES(10892977,16001,NULL,1,60,9,110435,x'0200',NULL,NULL,NULL,NULL,NULL,
> INSERT INTO ACHANGE(Z_PK,Z_ENT,Z_OPT,ZCHANGETYPE,ZENTITY,ZENTITYPK,ZTRANSACTIONID,ZCOLUMNS,ZTOMBSTONE0,ZTOMBSTONE1,ZTOMBSTONE2,ZTOMBSTONE3,ZTOMBSTONE4,ZTOMBSTONE5) VALUES(10892978,16001,NULL,1,60,5,110436,x'0300',NULL,NULL,NULL,NULL,NULL,
> INSERT INTO ATRANSACTION(Z_PK,Z_ENT,Z_OPT,ZAUTHORTS,ZBUNDLEIDTS,ZCONTEXTNAMETS,ZPROCESSIDTS,ZTIMESTAMP,ZAUTHOR,ZBUNDLEID,ZCONTEXTNAME,ZPROCESSID,ZQUERYGEN) VALUES(110434,16002,NULL,1,2,53,3,695780503.207048,NULL,NULL,NULL,NULL,x'18e22d000
> INSERT INTO ATRANSACTION(Z_PK,Z_ENT,Z_OPT,ZAUTHORTS,ZBUNDLEIDTS,ZCONTEXTNAMETS,ZPROCESSIDTS,ZTIMESTAMP,ZAUTHOR,ZBUNDLEID,ZCONTEXTNAME,ZPROCESSID,ZQUERYGEN) VALUES(110435,16002,NULL,1,2,53,3,695780847.228457,NULL,NULL,NULL,NULL,x'18e22d000
> INSERT INTO ATRANSACTION(Z_PK,Z_ENT,Z_OPT,ZAUTHORTS,ZBUNDLEIDTS,ZCONTEXTNAMETS,ZPROCESSIDTS,ZTIMESTAMP,ZAUTHOR,ZBUNDLEID,ZCONTEXTNAME,ZPROCESSID,ZQUERYGEN) VALUES(110436,16002,NULL,1,2,53,3,695780866.65678,NULL,NULL,NULL,NULL,x'18e22d0000
> UPDATE ZUSERFEEDBACK SET Z_OPT=3, ZTYPE=3, ZLASTMODIFIEDDATE=695780866.649606 WHERE Z_PK=5;                                                                                                                                                   
> UPDATE ZUSERFEEDBACK SET Z_OPT=4, ZTYPE=3, ZLASTMODIFIEDDATE=695780847.224094 WHERE Z_PK=9;                                                                                                                                                   
> UPDATE Z_PRIMARYKEY SET Z_MAX=10892978 WHERE Z_ENT=16001;                                                                                                                                                                                     
> UPDATE Z_PRIMARYKEY SET Z_MAX=110436 WHERE Z_ENT=16002;       
>

**Reply by RhetTbull** [reply]
Thanks -- will take a look. Looks like this is being set in `ZUSERFEEDBACK` which is a table I've not explored.  

```sql
CREATE TABLE ZUSERFEEDBACK (
     Z_PK INTEGER PRIMARY KEY,
     Z_ENT INTEGER,
     Z_OPT INTEGER,
     ZFEATURE INTEGER,
     ZTYPE INTEGER,
     ZMEMORY INTEGER,
     ZPERSON INTEGER,
     ZLASTMODIFIEDDATE TIMESTAMP,
     ZCONTEXT VARCHAR,
     ZCLOUDLOCALSTATE INTEGER,
     ZUUID VARCHAR,
     ZCLOUDDELETESTATE INTEGER
 );
```

**Reply by RhetTbull** [reply]
And interestingly, my Photos library has nothing in that table:

```sh
$ sqlite3 ~/Pictures/Photos\ Library.photoslibrary/database/Photos.sqlite ".headers on" "select * from ZUSERFEEDBACK"
$
```

**Reply by eecue** [reply]
Find a face Photos has recognized, tag it with something random like Test Person, go into People, view the person, at the bottom you’ll see the option to show them less.

**Reply by eecue** [reply]
on ios it gives your the option to reset their Feature Settings but os x you have to reset for everyone in the settings panel.

**Reply by RhetTbull** [reply]
@eecue what version of macOS are you using?  In my testing, the option shows up in People as "Feature This Person Less" and the ZUSERFEEDBACK.ZTYPE value is 2 where as in your example it's 3.

**Reply by RhetTbull** [reply]
Iv'e not merged the branch yet but have this feature implemented. I went with `ZTYPE != 0' as the value to indicate "feature this person less" as that was consistent in my Ventura testing.  There might be other values that mean different things but I've not been able to produce any yet.

**Comment by eecue** [comment]
I’m looking on my phone and I see there are two options: Feature this person less and Never feature this person. Either way your solution sounds perfect. On Jan 22, 2023, at 10:18 AM, Rhet Turnbull ***@***.***> wrote:﻿
Iv'e not merged the branch yet but have this feature implemented. I went with `ZTYPE != 0' as the value to indicate "feature this person less" as that was consistent in my Ventura testing.  There might be other values that mean different things but I've not been able to produce any yet.

**Reply by RhetTbull** [reply]
Added `PersonInfo.favorite` and `PersonInfo.feature_less` in [v0.56.6](https://github.com/RhetTbull/osxphotos/releases/tag/v0.56.6)
